addOns/wappalyzer/src/main/java/org/zaproxy/zap/extension/wappalyzer/WappalyzerPassiveScanner.java

```diff
@@ -165,6 +166,38 @@ private void checkMetaElementsMatches(Source source) {
         }
     }
 
+    private void checkDomElementMatches(Source source) {
+        Document doc = Jsoup.parse(String.valueOf(source));
```

Comment:
It would be better to pass in `HttpMessage msg`, then use `msg.getResponseBody().toString()`.

Reply:
This wasn't changed, or was reverted at some point. Source is a jericho class, it's already derived from the msg. Since we're specifically not using jericho here I believe it makes more sense to pass the message and access the body's toString method. Ex:

```java
    private void checkDomElementMatches(HttpMessage message) {
        Document doc = Jsoup.parse(message.getResponseBody().toString());
```